operator/cmd/mesh/operator-common.go

```diff
@@ -55,7 +55,7 @@ const (
 // isControllerInstalled reports whether an operator deployment exists in the given namespace.
 func isControllerInstalled(cs kubernetes.Interface, operatorNamespace string, revision string) (bool, error) {
 	orn := operatorResourceName
-	if revision != "" {
+	if revision != "" && revision != "default" {
```

Comment:
How it can be `default`? I thought we explicitly set `default` to `` in all of these kind of commands.

Reply:
If `orArgs.revision == ""` https://github.com/istio/istio/blob/master/operator/cmd/mesh/operator-remove.go#L87,
all revision will be hit.

Reply:
When no revision is specified we want only the `default` revision in `istioctl operator remove`, so special handling of default revision is required here.

Reply:
The default revision should be the same as non-revision, so we only need to handle `if revision != ""`. I think the original logic is slightly flawed. We should not tie 'purge' and 'revision' together. Instead, we only need to warn users that all revisions will be purged if they specify both flags, rather than disallowing them from setting the flags. This should be similar to the 'uninstall purge' scenario.

Reply:
If default is not processed here, according to the previous processing, the final deployment name to be processed will be `istio-operator-default`.
And if the previous processing sets the default revisions to "", then all revisions will be hit, it's not we expect.

Reply:
I think the expected behavior is: when we init, no matter `init` or `init --revision default`, it should all be `istio-operator`

Reply:
Yes, but when deleting now, if you don’t know the revision or revision == `default`, all revisions will be cleaned up:https://github.com/istio/istio/issues/45242#issue-1736615170